semantic-conventions/src/opentelemetry/semconv/model/semantic_convention.py

```diff
@@ -238,7 +251,35 @@ def __init__(self, group):
 class MetricSemanticConvention(BaseSemanticConvention):
     GROUP_TYPE_NAME = "metric"
 
-    allowed_keys = ()
+    allowed_keys: Tuple[str, ...] = BaseSemanticConvention.allowed_keys + ("metrics",)
+
+    class Metric:
+        def __init__(self, metric, parent_prefix):
+            self.id: str = metric.get("id")
+            self.fqn = "{}.{}".format(parent_prefix, self.id)
+            self.instrument: InstrumentKind = InstrumentKind[metric.get("instrument")]
+            self.units: str = metric.get("units")
+            self.brief: str = metric.get("brief")
+
+            if None in [metric.get("instrument"), self.id, self.units, self.brief]:
+                raise ValueError
+
+    def __init__(self, group):
+        super().__init__(group)
+        self.metrics = ()
+        if group.get("metrics"):
+            try:
+                self.metrics: Tuple[MetricSemanticConvention.Metric] = tuple(
+                    map(
+                        lambda m: MetricSemanticConvention.Metric(m, self.prefix),
+                        group.get("metrics"),
+                    )
+                )
+            except ValueError as e:
+                raise ValidationError.from_yaml_pos(
```

Comment:
I think it would be clearer if this error was directly thrown from `Metric.__init__`.

Reply:
Agreed, I did it here because `Metric` doesn't have access to `_position`. Anyway, I have just passed `_position` into the constructor, so I can raise from there now.